### DISCUSSION TITLE
How should people contribute to this project?

### DISCUSSION THREAD

**Discussion by RonSijm** [MV-GH/lemmy_openapi_spec#12] [General]
Hey there,

I started using the specs, and I was curious how people should be contributing. As I'm generating my client and am using it, I'm noticing some things are slightly incorrect, or missing.

As I understand it correctly, the [lemmy_spec.yaml](https://github.com/MV-GH/lemmy_openapi_spec/blob/master/lemmy_spec.yaml) file is auto-generated, right? 

So I suppose it wouldn't really make sense to send PRs with fixes directly to the specs, right? Or is that still useful to know what the end result should be or something?

**Comment by MV-GH** [comment]
I have a `CONTRIBUTING.md` which sheds some light on the internals and what is autogenerated. I shall add a link to this in the README.md for better visibility

**Reply by RonSijm** [reply]
Hehe, yea, I saw the `CONTRIBUTING.md` - and it has a chapter of  `How does it work` and `How to run it` - 

But not really info on how to contribute or what to do if I'm noticing anything incorrect in the spec file 🙈

**Reply by MV-GH** [reply]
Alright If you want to modify the non automated parts of the spec than you will have to modify `api_routes.yaml`
For the automated parts you would have to update the relevant script in `/src`

**Reply by MV-GH** [reply]
I have updated the CONTRIBUTING.md to include a section for that
